HOT Movies/HOT Movies/Scenes/Movie Deltail Screen/Similar Movie Cell/SimilarMoviesTableViewCell.swift

```diff
@@ -0,0 +1,81 @@
+//
+//  SimilarMoviesTableViewCell.swift
+//  HOT Movies
+//
+//  Created by Thuận Nguyễn Văn on 21/07/2021.
+//
+
+import UIKit
+import Reusable
+import RxCocoa
+import RxSwift
+import NSObject_Rx
+
+protocol SimilarMoviesTableViewCellDelegate: class {
+    
+    func pushDetailsViewController(movie: Movie)
+}
```

Comment:
dùng closure cho gọn cũng được

Reply:
Bên này nó gọn đi nhưng bên Màn Deltail lại phình to code lên ở chỗ tạo cell anh ạ. Em vừa code lại thấy nó không clean lắm.

Reply:
Tại vì cái này nó còn phải tạo lại useCase, viewModel, ... Nên bên kia cũng thêm 5 6 dòng mà code ngay vào chỗ tạo xem em thấy nó hơi rối mắt anh ạ.

Reply:
`            case .similar(let model):
                let cell = tableView.dequeueReusableCell(for: indexPath, cellType: SimilarMoviesTableViewCell.self)
                cell.similarMovieCellTapped = { [weak self] movie in
                    let viewController = MovieDetailsViewController()
                    let useCase = MovieDetailsUseCase(movieRepository: MoviesRepository())
                    let viewModel = MovieDetailsViewModel(useCase: useCase, movie: movie)
                    viewController.bindViewModel(to: viewModel)
                    self?.navigationController?.pushViewController(viewController, animated: true)
                }
                cell.configureCell(movies: model)
                return cell`

Reply:
case .similar(let model):
                let cell = tableView.dequeueReusableCell(for: indexPath, cellType: SimilarMoviesTableViewCell.self)
                cell.similarMovieCellTapped = { [weak self] movie in
                    let viewController = MovieDetailsViewController()
                    let useCase = MovieDetailsUseCase(movieRepository: MoviesRepository())
                    let viewModel = MovieDetailsViewModel(useCase: useCase, movie: movie)
                    viewController.bindViewModel(to: viewModel)
                    self?.navigationController?.pushViewController(viewController, animated: true)
                }
                cell.configureCell(movies: model)
                return cell

Reply:
Em để nguyên anh nha :D

Reply:
việc chuyển màn hình là do navigator xử lý, viewcontroller đâu có khởi tạo mấy cái này đâu em

Reply:
Vâng em sửa rồi ạ.